server/container/guice/guice-common/src/test/java/org/apache/james/JamesServerExtension.java

```diff
@@ -92,16 +96,22 @@ public void afterAll(ExtensionContext extensionContext) throws Exception {
 
     @Override
     public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-        return (parameterContext.getParameter().getType() == GuiceJamesServer.class)
-            || registrableExtension.supportsParameter(parameterContext, extensionContext);
+        return JamesParametersResolver.builder()
+            .jamesServer(guiceJamesServer)
+            .registrableExtension(registrableExtension)
+            .paramsRegistration(paramsRegistration)
+            .build()
```

Comment:
This can be done in the constructor?

Reply:
not really, `guiceJamesServer` is mutable, it's assigned to a new instance before each test.